File: Python/String-3/sum_digits.py

# given a string, return the sum of the digits 0-9 that appear in the string, ignoring all other characters
# return 0 if there are no digits in the string
def sum_digits(str):
    sum = 0

    for x in range(len(str)):
        if str[x].isdigit():
            sum = sum + int(str[x])

    return sum

print(sum_digits('aa1bc2d3'))
print(sum_digits('aa11b33'))
print(sum_digits('Chocolate'))

File: Java/String-3/sumDigits.java

public class string3 {

	// given a string, return the sum of the digits 0-9 that appear in the string, ignoring all other characters
	// return 0 if there are no digits in the string
	public int sumDigits(String str) {
		
		int sum = 0;

		for (int i = 0; i < str.length(); i++) {
			if (Character.isDigit(str.charAt(i))) {
				sum = sum + Integer.parseInt(str.substring(i, i+1));
			}
		}
		
		return sum; 
		
	}	
	
	public static void main(String[] args) {
		
		string3 o = new string3();
		
		System.out.println(o.sumDigits("aa1bc2d3"));
		System.out.println(o.sumDigits("aa11b33"));
		System.out.println(o.sumDigits("Chocolate"));

	}

}
